sympy/functions/special/beta_functions.py

```diff
@@ -163,7 +163,7 @@ def _eval_rewrite_as_gamma(self, x, y, piecewise=True, **kwargs):
 
     def _eval_rewrite_as_Integral(self, x, y, **kwargs):
         from sympy.integrals.integrals import Integral
-        t = Dummy('t')
+        t = uniquely_named_symbol('t').as_dummy()
```

Comment:
This will not help as you will get exactly the same expression as earlier. You need to pass the other involved expressions using the `expr`  argument.

Reply:
```suggestion
        t = Dummy(uniquely_named_symbol('t', [x, y]).name)
```